README.md
# EventDB Module for Icinga Web 2

[![Build Status](https://travis-ci.org/Icinga/icingaweb2-module-eventdb.svg?branch=master)](https://travis-ci.org/Icinga/icingaweb2-module-eventdb)
[![Github Tag](https://img.shields.io/github/tag/Icinga/icingaweb2-module-eventdb.svg)](https://github.com/Icinga/icingaweb2-module-eventdb)

![Icinga Logo](https://www.icinga.com/wp-content/uploads/2014/06/icinga_logo.png)

* [About](#about)
* [Requirements](#requirements)
* [Getting Started](#getting-started)
* [Documentation](#documentation)
* [License](#license)

## About

With the EventDB Module you can browse, comment and acknowledge events collected
by [EventDB](https://git.netways.org/eventdb/eventdb) easily in
[Icinga Web 2](https://www.icinga.org/products/icinga-web-2/).

![screenshot](doc/screenshots/overview.png)

Also see [Screenshots](doc/10-Screenshots.md) in documentation.

## Requirements

* Icinga Web 2
* A database with events collected by EventDB

## Getting started

Install and enable the module, then go to [Configuration](doc/02-Configuration.md)
to set up the essentials.

## Documentation

* [Configuration](doc/02-Configuration.md)
* [Custom variables](doc/03-CustomVars.md)
* [Security](doc/09-Security.md)
* [Screenshots](doc/10-Screenshots.md)

## License

    Copyright (C) 2016-2017 Icinga Development Team
                  2016-2017 Eric Lippmann <eric.lippmann@icinga.com>
                  2017      Markus Frosch <markus.frosch@icinga.com>

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


CHANGELOG.md
# Change Log

## [v1.3.0](https://github.com/Icinga/icingaweb2-module-eventdb/tree/v1.3.0) (2018-01-25)
[Full Changelog](https://github.com/Icinga/icingaweb2-module-eventdb/compare/v1.2.0...v1.3.0)

**Implemented enhancements:**

- Links in Event text [\#24](https://github.com/Icinga/icingaweb2-module-eventdb/issues/24)
- Severity Widget for fast clicking users [\#23](https://github.com/Icinga/icingaweb2-module-eventdb/issues/23)
- Add default\_filter config option for the menu url [\#27](https://github.com/Icinga/icingaweb2-module-eventdb/pull/27) ([lazyfrosch](https://github.com/lazyfrosch))
- Add DetailviewExtensionHook for detail and multi-select [\#26](https://github.com/Icinga/icingaweb2-module-eventdb/pull/26) ([lazyfrosch](https://github.com/lazyfrosch))

**Fixed bugs:**

- Detailview extension shown despite customvar not present [\#25](https://github.com/Icinga/icingaweb2-module-eventdb/issues/25)
- Pagination issue [\#22](https://github.com/Icinga/icingaweb2-module-eventdb/issues/22)

## [v1.2.0](https://github.com/Icinga/icingaweb2-module-eventdb/tree/v1.2.0) (2017-08-17)
[Full Changelog](https://github.com/Icinga/icingaweb2-module-eventdb/compare/v1.1.0...v1.2.0)

**Implemented enhancements:**

- Add more output / format options [\#21](https://github.com/Icinga/icingaweb2-module-eventdb/pull/21) ([lazyfrosch](https://github.com/lazyfrosch))

## [v1.1.0](https://github.com/Icinga/icingaweb2-module-eventdb/tree/v1.1.0) (2017-08-16)
[Full Changelog](https://github.com/Icinga/icingaweb2-module-eventdb/compare/v1.0.0...v1.1.0)

**Implemented enhancements:**

- Add screenshot to README.md [\#1](https://github.com/Icinga/icingaweb2-module-eventdb/issues/1)
- Support EDBC extensions [\#15](https://github.com/Icinga/icingaweb2-module-eventdb/issues/15)
- Event list should be auto-reloading [\#13](https://github.com/Icinga/icingaweb2-module-eventdb/issues/13)
- Add integration to host and service [\#12](https://github.com/Icinga/icingaweb2-module-eventdb/issues/12)
- Add German translation [\#9](https://github.com/Icinga/icingaweb2-module-eventdb/issues/9)
- Popup Tool Tips for Action Buttons in Interface missing  [\#7](https://github.com/Icinga/icingaweb2-module-eventdb/issues/7)
- Fuzzy link to hostnames [\#5](https://github.com/Icinga/icingaweb2-module-eventdb/issues/5)
- Add EDBC extensions [\#20](https://github.com/Icinga/icingaweb2-module-eventdb/pull/20) ([lazyfrosch](https://github.com/lazyfrosch))
- Add detailview extension [\#19](https://github.com/Icinga/icingaweb2-module-eventdb/pull/19) ([lazyfrosch](https://github.com/lazyfrosch))
- Integration into monitoring [\#16](https://github.com/Icinga/icingaweb2-module-eventdb/pull/16) ([lazyfrosch](https://github.com/lazyfrosch))
- Improve list and detail UI [\#14](https://github.com/Icinga/icingaweb2-module-eventdb/pull/14) ([lazyfrosch](https://github.com/lazyfrosch))
- Improve usability of priority/severity widget [\#11](https://github.com/Icinga/icingaweb2-module-eventdb/pull/11) ([lazyfrosch](https://github.com/lazyfrosch))
- DB repository: add convert rules for event.host\_address [\#6](https://github.com/Icinga/icingaweb2-module-eventdb/pull/6) ([Al2Klimov](https://github.com/Al2Klimov))

**Fixed bugs:**

- Acknowledgement only available in multi selection [\#8](https://github.com/Icinga/icingaweb2-module-eventdb/issues/8)

**Closed issues:**

- Configuration for extra types [\#18](https://github.com/Icinga/icingaweb2-module-eventdb/issues/18)
- Add Host Column to Overview [\#2](https://github.com/Icinga/icingaweb2-module-eventdb/issues/2)
- The content of 'Host Address' is not readable [\#4](https://github.com/Icinga/icingaweb2-module-eventdb/issues/4)

**Merged pull requests:**

- Add basic phpunit environment [\#17](https://github.com/Icinga/icingaweb2-module-eventdb/pull/17) ([lazyfrosch](https://github.com/lazyfrosch))
- Add GitHub issue template [\#10](https://github.com/Icinga/icingaweb2-module-eventdb/pull/10) ([dnsmichi](https://github.com/dnsmichi))



\* *This Change Log was automatically generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator)*

library/Eventdb/Event.php
<?php
/* Icinga Web 2 | (c) 2016 Icinga Development Team | GPLv2+ */

namespace Icinga\Module\Eventdb;

use ArrayObject;

class Event extends ArrayObject
{
    public static $facilities = array(
        0 => 'kernel messages',
        1 => 'user-level messages',
        2 => 'mail system',
        3 => 'system daemons',
        4 => 'security/authorization messages',
        5 => 'messages generated internally by syslogd',
        6 => 'line printer subsystem',
        7 => 'network news subsystem',
        8 => 'UUCP subsystem',
        9 => 'clock daemon',
        10 => 'security/authorization messages',
        11 => 'FTP daemon',
        12 => 'NTP subsystem',
        13 => 'log audit',
        14 => 'log alert',
        15 => 'clock daemon',
        16 => 'local use 0',
        17 => 'local use 1',
        18 => 'local use 2',
        19 => 'local use 3',
        20 => 'local use 4',
        21 => 'local use 5',
        22 => 'local use 6',
        23 => 'local use 7'
    );

    public static $priorities = array(
        0 => 'emergency',
        1 => 'alert',
        2 => 'critical',
        3 => 'error',
        4 => 'warning',
        5 => 'notice',
        6 => 'info',
        7 => 'debug'
    );

    public static $types = array(
        0 => 'syslog',
        1 => 'snmp',
        2 => 'mail'
    );

    public static $typeIcons = array(
        '_default' => 'help',
        'syslog'   => 'doc-text',
        'snmp'     => 'plug',
        'mail'     => 'bell',
    );

    public function __construct($data)
    {
        parent::__construct($data, ArrayObject::ARRAY_AS_PROPS);
    }

    public function offsetGet($index)
    {
        if (! $this->offsetExists($index)) {
            return null;
        }
        $getter = 'get' . ucfirst($index);
        if (method_exists($this, $getter)) {
            return $this->$getter();
        }
        return parent::offsetGet($index);
    }

    public function getAck()
    {
        return (bool) parent::offsetGet('ack');
    }

    public function getFacility()
    {
        $facility = (int) parent::offsetGet('facility');
        return array_key_exists($facility, static::$facilities) ? static::$facilities[$facility] : $facility;
    }

    public function getPriority()
    {
        $priority = (int) parent::offsetGet('priority');
        return array_key_exists($priority, static::$priorities) ? static::$priorities[$priority] : $priority;
    }

    public function getType()
    {
        $type = (int) parent::offsetGet('type');
        return array_key_exists($type, static::$types) ? static::$types[$type] : $type;
    }

    public function getTypeIcon()
    {
        if (array_key_exists($type = $this->getType(), static::$typeIcons)) {
            return static::$typeIcons[$type];
        } else {
            return static::$typeIcons['_default'];
        }
    }

    public static function fromData($data)
    {
        return new static($data);
    }

    public static function getPriorityId($priorityName)
    {
        $priorities = array_flip(static::$priorities);
        return $priorities[$priorityName];
    }
}
